script/wait_for_file.sh

```diff
@@ -0,0 +1,8 @@
+wait_for_file() {
+  local file="$1"; shift
+  local wait_seconds="${1:-10}"; shift # 10 seconds as default timeout
+
+  until test $((wait_seconds--)) -eq 0 -o -f "$file" ; do sleep 1; done
```

Comment:
Small race condition here - if file appears on last iteration, you'd miss this, effectively waiting only N-1 seconds for the file to appear. But probably not worth it complicating the loop just to catch that.